superset/models/database.py

```diff
@@ -0,0 +1,601 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import json
+import logging
+import textwrap
+from contextlib import closing
+from copy import deepcopy
+from typing import Any, Callable, Dict, List, Optional, Set, Tuple, Type
+
+# pylint: disable=ungrouped-imports
+import numpy
+import pandas as pd
+import sqlalchemy as sqla
+import sqlparse
+from flask import g, request
+from flask_appbuilder import Model
+from sqlalchemy import (
+    Boolean,
+    Column,
+    create_engine,
+    Integer,
+    MetaData,
+    String,
+    Table,
+    Text,
+    UniqueConstraint,
+)
+from sqlalchemy.engine import Dialect, Engine, url
+from sqlalchemy.engine.reflection import Inspector
+from sqlalchemy.engine.url import make_url, URL
+from sqlalchemy.pool import NullPool
+from sqlalchemy.sql import Select
+from sqlalchemy_utils import EncryptedType
+
+from superset import db_engine_specs, security_manager
+from superset.db_engine_specs.base import TimeGrain
+from superset.models.core import (
+    config,
+    custom_password_store,
```

Comment:
To pass the failing unit test, I think the function definition for `custom_password_store` should be brought in from `superset.models.core` to this very location, and the test should target this function definition here in this module after it gets moved.

Ultimately we need to improve the `config` handling in tests (allowing tests to temporary alter configs during the scope of a unit test), but in the meantime for the sake of this PR, moving the definition and test monkey patch is probably ok.

Reply:
More generally here for the other imports out of `superset.models.core`, I think they are probably only used here, so instead of importing them maybe just defining them here if possible...

Reply:
oh good catch, thanks